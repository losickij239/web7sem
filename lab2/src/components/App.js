import React from 'react';
import {Provider} from 'react-redux';
import 'bootstrap/dist/css/bootstrap.css'
import LargeCityInfo from "./LargeCityInfo";
import FavoriteCityPack from './FavoriteCityPack'
import configureStore from '../store/configureStore';


const store = configureStore();

const favCity =
    [
        {
            'city': 'moscow',
            'state': {
                "coord": {
                    "lon": 37.62,
                    "lat": 55.75
                },
                "weather": [
                    {
                        "id": 800,
                        "main": "Clear",
                        "description": "clear sky",
                        "icon": "01n"
                    }
                ],
                "base": "stations",
                "main": {
                    "temp": 27.37,
                    "pressure": 1017,
                    "humidity": 86,
                    "temp_min": 276.48,
                    "temp_max": 278.15
                },
                "visibility": 10000,
                "wind": {
                    "speed": 2,
                    "deg": 240
                },
                "clouds": {
                    "all": 0
                },
                "dt": 1569349865,
                "sys": {
                    "type": 1,
                    "id": 9029,
                    "message": 0.0065,
                    "country": "RU",
                    "sunrise": 1569295059,
                    "sunset": 1569338749
                },
                "timezone": 10800,
                "id": 524901,
                "name": "Moscow",
                "cod": 200
            }
        },
        {
            'city': 'kiev',
            'state': {
                "coord": {
                    "lon": 37.62,
                    "lat": 55.75
                },
                "weather": [
                    {
                        "id": 800,
                        "main": "Clear",
                        "description": "clear sky",
                        "icon": "01n"
                    }
                ],
                "base": "stations",
                "main": {
                    "temp": 277.37,
                    "pressure": 1017,
                    "humidity": 86,
                    "temp_min": 276.48,
                    "temp_max": 278.15
                },
                "visibility": 10000,
                "wind": {
                    "speed": 2,
                    "deg": 240
                },
                "clouds": {
                    "all": 0
                },
                "dt": 1569349865,
                "sys": {
                    "type": 1,
                    "id": 9029,
                    "message": 0.0065,
                    "country": "RU",
                    "sunrise": 1569295059,
                    "sunset": 1569338749
                },
                "timezone": 10800,
                "id": 524901,
                "name": "Moscow",
                "cod": 200
            }
        },
        {
            'city': 'piter',
            'state': {
                "coord": {
                    "lon": 37.62,
                    "lat": 55.75
                },
                "weather": [
                    {
                        "id": 800,
                        "main": "Clear",
                        "description": "clear sky",
                        "icon": "01n"
                    }
                ],
                "base": "stations",
                "main": {
                    "temp": 277.37,
                    "pressure": 1017,
                    "humidity": 86,
                    "temp_min": 276.48,
                    "temp_max": 278.15
                },
                "visibility": 10000,
                "wind": {
                    "speed": 2,
                    "deg": 240
                },
                "clouds": {
                    "all": 0
                },
                "dt": 1569349865,
                "sys": {
                    "type": 1,
                    "id": 9029,
                    "message": 0.0065,
                    "country": "RU",
                    "sunrise": 1569295059,
                    "sunset": 1569338749
                },
                "timezone": 10800,
                "id": 524901,
                "name": "Moscow",
                "cod": 200
            }
        },
        {
            'city': 'catcity',
            'state': {
                "coord": {
                    "lon": 37.62,
                    "lat": 55.75
                },
                "weather": [
                    {
                        "id": 800,
                        "main": "Clear",
                        "description": "clear sky",
                        "icon": "01n"
                    }
                ],
                "base": "stations",
                "main": {
                    "temp": 277.37,
                    "pressure": 1017,
                    "humidity": 86,
                    "temp_min": 276.48,
                    "temp_max": 278.15
                },
                "visibility": 10000,
                "wind": {
                    "speed": 2,
                    "deg": 240
                },
                "clouds": {
                    "all": 0
                },
                "dt": 1569349865,
                "sys": {
                    "type": 1,
                    "id": 9029,
                    "message": 0.0065,
                    "country": "RU",
                    "sunrise": 1569295059,
                    "sunset": 1569338749
                },
                "timezone": 10800,
                "id": 524901,
                "name": "Moscow",
                "cod": 200
            }
        },
    ];


function App() {
    return (
        <div className="container">
            <div className="jumbotron">
                <h1 className="display-2">Погода для тебя</h1>
            </div>
            <LargeCityInfo/>
            <Provider store={store}>
                <FavoriteCityPack/>
            </Provider>
        </div>
    )
}


export default App