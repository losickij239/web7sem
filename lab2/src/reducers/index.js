import { combineReducers } from 'redux';
import { items, newCityValue } from './items';

export default combineReducers({
    items,
    newCityValue
});