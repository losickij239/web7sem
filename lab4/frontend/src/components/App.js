import React from 'react';
import {Provider} from 'react-redux';
import 'bootstrap/dist/css/bootstrap.css'
import LargeCityInfo from "./LargeCityInfo";
import FavoriteCityPack from './FavoriteCityPack'
import configureStore from '../store/configureStore';


const store = configureStore();

function App() {
    return (
        <div className="container">
            <div className="jumbotron">
                <h1 className="display-2">Погода для тебя</h1>
            </div>
            <LargeCityInfo/>
            <Provider store={store}>
                    <FavoriteCityPack/>
            </Provider>
        </div>
    )
}


export default App