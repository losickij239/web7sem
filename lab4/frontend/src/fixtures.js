export default {
    "coord": {
        "lon": 37.62,
        "lat": 55.75
    },
    "weather": [
        {
            "id": 800,
            "main": "Clear",
            "description": "clear sky",
            "icon": "01n"
        }
    ],
    "base": "stations",
    "main": {
        "temp": 277.37,
        "pressure": 1017,
        "humidity": 86,
        "temp_min": 276.48,
        "temp_max": 278.15
    },
    "visibility": 10000,
    "wind": {
        "speed": 2,
        "deg": 240
    },
    "clouds": {
        "all": 0
    },
    "dt": 1569349865,
    "sys": {
        "type": 1,
        "id": 9029,
        "message": 0.0065,
        "country": "RU",
        "sunrise": 1569295059,
        "sunset": 1569338749
    },
    "timezone": 10800,
    "id": 524901,
    "name": "Moscow",
    "cod": 200
}