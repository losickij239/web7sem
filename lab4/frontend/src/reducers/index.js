import { combineReducers } from 'redux';
import { items, newCityValue, errorHandler } from './items';

export default combineReducers({
    items,
    newCityValue,
    errorHandler
});