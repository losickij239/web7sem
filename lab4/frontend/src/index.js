import React from 'react';
import {render} from 'react-dom'
import App from './components/App'


fetch('http://localhost:3001/')
    .then(() => {
            render(<App/>, document.getElementById('root'))
        }
    )
    .catch(() => {
        render(<div className="col col-6 alert alert-danger">Бекенд недоступен, попробуйте
            позднее </div>, document.getElementById('root'));
    });


