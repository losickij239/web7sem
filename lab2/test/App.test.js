import React from 'react';
import { shallow } from '../src/enzyme';
import { shallowToJson  } from 'enzyme-to-json';
import App  from '../src/components/App';

it('App отображается корректно', () => {
    const fullApp = shallow(
        <App />
    );
    expect(shallowToJson(fullApp)).toMatchSnapshot();
});
