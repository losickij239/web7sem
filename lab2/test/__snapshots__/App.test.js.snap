// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App отображается корректно 1`] = `
<div
  className="container"
>
  <div
    className="jumbotron"
  >
    <h1
      className="display-2"
    >
      Погода для тебя
    </h1>
  </div>
  <LargeCityInfo />
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <PersistGate
      loading={null}
      persistor={
        Object {
          "dispatch": [Function],
          "flush": [Function],
          "getState": [Function],
          "pause": [Function],
          "persist": [Function],
          "purge": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <Connect(FavoriteCityPack) />
    </PersistGate>
  </Provider>
</div>
`;
